test_xdb/models/under_test/timeadd_test.sql

```diff
@@ -1,15 +1,25 @@
 WITH
 source_data AS (
-    SELECT '2020-01-01 00:00:00' AS date_col
+    SELECT 
+        '2020-01-01 00:00:00' AS date_col_string
+        , '2020-01-01 00:00:00'::TIMESTAMP AS date_col_timestamp
 )
 
 SELECT
-    {{xdb.timeadd('second', '1', 'date_col')}} AS one_second_added
-    , {{xdb.timeadd('second', '-1', 'date_col')}} AS one_second_subtracted
-    , {{xdb.timeadd('minute', '1', 'date_col')}} AS one_minute_added
-    , {{xdb.timeadd('minute', '-1', 'date_col')}} AS one_minute_subtracted
-    , {{xdb.timeadd('hour', '1', 'date_col')}} AS one_hour_added
-    , {{xdb.timeadd('hour', '-1', 'date_col')}} AS one_hour_subtracted
+--cast to timestamp logic block check
+    {{xdb.timeadd('second', '1', 'date_col_string')}} AS one_second_added
+    , {{xdb.timeadd('second', '-1', 'date_col_string')}} AS one_second_subtracted
+    , {{xdb.timeadd('minute', '1', 'date_col_string')}} AS one_minute_added
+    , {{xdb.timeadd('minute', '-1', 'date_col_string')}} AS one_minute_subtracted
+    , {{xdb.timeadd('hour', '1', 'date_col_string')}} AS one_hour_added
+    , {{xdb.timeadd('hour', '-1', 'date_col_string')}} AS one_hour_subtracted
+--NO cast to timestamp logic block check
+    , {{xdb.timeadd('second', '1', 'date_col_timestamp', false)}} AS no_cast_one_second_added
+    , {{xdb.timeadd('second', '-1', 'date_col_timestamp', false)}} AS no_cast_one_second_subtracted
+    , {{xdb.timeadd('minute', '1', 'date_col_timestamp', false)}} AS no_cast_one_minute_added
+    , {{xdb.timeadd('minute', '-1', 'date_col_timestamp', false)}} AS no_cast_one_minute_subtracted
+    , {{xdb.timeadd('hour', '1', 'date_col_timestamp', false)}} AS no_cast_one_hour_added
+    , {{xdb.timeadd('hour', '-1', 'date_col_timestamp', false)}} AS no_cast_one_hour_subtracted
 FROM
     source_data
 
```

Comment:
If I remember how xdb tests work I think there should be a corresponding yml file in https://github.com/Health-Union/dbt-xdb/tree/master/test_xdb/models/schema_tests with the accepted values for this query. I don't see one though. So maybe we never had one. Could you create one?

Reply:
I've found these test properties in [schema.yml](https://github.com/Health-Union/dbt-xdb/blob/master/test_xdb/models/schema_tests/schema.yml), but you're definitely right about absence of etalon values. Instead of adding of new columns to result table of the test I've changed a bit its query. This change allows to avoid updates of [schema.yml](https://github.com/Health-Union/dbt-xdb/blob/master/test_xdb/models/schema_tests/schema.yml) then.

Reply:
Sorry if this is just my ignorance, but how would this test work? How would it be run and what would the results look like?

Reply:
The test suite generates a test model and issue a dbt test command against those models to verify the values that the model generates.

In this case, we're adding another row to the existing model, so the dbt test set should still pass since the output values should be the same (for each column).

For most of the newer test models, the dbt tests were moved out of the `schema.yml` file and into individual `.yml` files for each test model.